tests/nlu/featurizers/test_lexical_syntactic_featurizer.py

```diff
@@ -70,6 +70,58 @@ def test_text_featurizer(sentence, expected_features):
     assert np.all(seq_vec.toarray() == expected_features[:-1])
 
 
+@pytest.mark.parametrize(
+    "sentence, expected_features",
+    [
+        (
+            "hello goodbye Goodbye",
+            [
+                [0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0,],
+                [0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 1.0,],
+                [1.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0,],
+            ],
+        ),
+        (
+            "a 1 A",
+            [
+                [0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0,],
+                [0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 1.0,],
+                [1.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0,],
+            ],
+        ),
+    ],
+)
+def test_text_featurizer_case_insensitive(sentence, expected_features):
```

Comment:
To test this functionality, the test should assert whether the features of "Goodbye" are identical to the features of "goodby"

Reply:
Reworked the test to check only for different features of the same token.lower()
in https://github.com/RasaHQ/rasa/pull/10482/commits/e2e36d5d05ecfdd6caddfe33c176c9018ae6e624